src/Controller/GameController.php

```diff
@@ -44,13 +45,15 @@ public function index(?string $action): string
 
         $levelManager = new LevelManager();
         $level = $levelManager->selectOneById($levelId);
-        $cells = LevelManager::parseContent($level['content']);
 
-        $this->move($cells, $action ?? "");
+        $tileManager = new TileManager();
+        $tiles = $tileManager->selectAllByLevelId($levelId);
+
+        $this->move($tiles, $action ?? "");
```

Comment:
you may just compute the potential new position, then search for a tile of the level at the potential position which is not a wall: if it exists, accept the move; otherwise keep the old position